### ISSUE TITLE
vixie cron email not accepted by gmail

### ISSUE THREAD

**Issue by rbrewer123** [NixOS/nixpkgs#70562]
**Describe the bug**
If the default nixos 19.03 cron (vixie cron) sends an email that is forwarded to a gmail account, it is rejected by gmail due to an invalid "From:" line.

**To Reproduce**
1. Set networking.defaultMailServer.root to forward to a gmail account.
2. Set up a cron job via services.cron.systemCronJobs.
3. email sent by cron is rejected by the gmail account.

**Expected behavior**
Cron emails should be properly forwarded to gmail.

**Additional context**
Forwarding cron emails to gmail used to work fine. It seems gmail has tightened their "From:" policy sometime in the last 6 months or so. It seems that the “From: root" is not acceptable to gmail anymore, but “From: root@host" is acceptable.

I have spent some time trying various things to fix this issue.  The following patch to cron seems to work:

```
diff -u cron_4.1.orig/do_command.c cron_4.1/do_command.c
--- cron_4.1.orig/do_command.c	2019-10-05 13:52:14.687920664 -0400
+++ cron_4.1/do_command.c	2019-10-05 14:50:08.095668180 -0400
@@ -417,7 +417,8 @@
 					perror(mailcmd);
 					(void) _exit(ERROR_EXIT);
 				}
-				fprintf(mail, "From: root (Cron Daemon)\n");
+				fprintf(mail, "From: Cron Daemon <root@%s>\n",
+                                        hostname);
 				fprintf(mail, "To: %s\n", mailto);
 				fprintf(mail, "Subject: Cron <%s@%s> %s\n",
 					usernm, first_word(hostname, "."),
```

**Comment by aanderse** [comment]
@rbrewer123 So obviously not a NixOS issue... but I don't think `cron` sees many releases these days so it is likely we are going to have to apply a fix here. Have you looked into how other distros are handling this problem?

**Comment by rbrewer123** [comment]
@aanderse The nicest solution I've seen is that some crons support a `MAILFROM` environment variable, analogous to `MAILTO`, which is simply filled in as the `From:` email address. NixOS could expose a `mailfrom` attribute on cron similar to the existing `mailto` attribute.

My patch above is sufficient but obviously not as flexible as `MAILFROM`.

**Comment by aanderse** [comment]
@rbrewer123 any motivation to work on this at all? Maybe grab a patch from another distro that implements the `MAILFROM` and run with that? It might take me a bit to get around to this so *if* you were able to pitch in I would **really** appreciate it.

**Comment by rbrewer123** [comment]
@aanderse Yeah, I'll get a `MAILFROM` patch ready for this.  Since this PR will need a man page update, I'd like to fold in some changes for #70906 too.

**Comment by aanderse** [comment]
Great! Thanks!

**Comment by aanderse** [comment]
thanks for your comments @AndersonTorres!

i created #452426 because it looks like `cronie` tries to write to the nix store... i'll play around with it a bit and see if `cronie` is a suitable replacement for `cron` in our module and report back

**Comment by aanderse** [comment]
from some minor experimentation it seems like `cronie` is a suitable drop in replacement for `cron`

the `cron` module would need to be extended to have a `MAILFROM` option though - anyone here motivated to tackle this?
